### ISSUE TITLE
Making spiketrains with more times than waveforms does not produce error

### ISSUE THREAD

**Issue by lepmik** [NeuralEnsemble/python-neo#270]
```python
neo.SpikeTrain(times=np.arange(4), waveforms=np.ones((10,6,50)), units='s', t_stop=4)
```
Shouldn't this produce an error since `len(times) != waveforms.shape[0]`

**Comment by apdavison** [comment]
Yes, this should produce an error.

(Note that it is planned in future to have `waveforms` be a 3D `AnalogSignal`)

**Comment by apdavison** [comment]
Fixed in #279
